ampd/src/lib.rs

```diff
@@ -188,18 +190,15 @@ where
 {
     #[allow(clippy::too_many_arguments)]
     fn new(
-        tm_client: tendermint_rpc::HttpClient,
+        event_publisher: event_sub::EventPublisher<tendermint_rpc::HttpClient>,
+        event_subscriber: event_sub::EventSubscriber,
```

Comment:
is this necessary?

Reply:
The grpc server needs a clone of the `event_subscriber`, so I had to initialize the event sub before this `new` is called. Any other suggestions?

Reply:
oh I searched for the usage and didn't find a hit, but it was just collapsed in unchanged code